presto-common/src/main/java/com/facebook/presto/common/type/TypeSignature.java

```diff
@@ -146,6 +146,25 @@ public List<TypeSignature> getTypeParametersAsTypeSignatures()
         return result;
     }
 
+    public List<TypeSignature> getTypeOrNamedTypeParametersAsTypeSignatures()
+    {
+        List<TypeSignature> result = new ArrayList<>();
+        for (TypeSignatureParameter parameter : parameters) {
+            if ((parameter.getKind() != ParameterKind.TYPE) &&
+                    (parameter.getKind() != ParameterKind.NAMED_TYPE)) {
```

Comment:
```suggestion
            if (!(parameter.getKind() == ParameterKind.TYPE &&
                    (parameter.getKind() == ParameterKind.NAMED_TYPE)) {
```

Reply:
@tdcmeehan I thought the original is easier to read. but your suggestion should have || instead of &&.

Reply:
Edited.  I read this as "if the parameter kind is not a TYPE OR a NAMED TYPE", which to me is just a more natural way of describing the outcome than "if the parameter is not a TYPE AND not a NAMED TYPE".

But I just realized we could just do this, which seems better:

```java
    public List<TypeSignature> getTypeOrNamedTypeParametersAsTypeSignatures()
    {
        {
            List<TypeSignature> result = new ArrayList<>();
            for (TypeSignatureParameter parameter : parameters) {
                switch (parameter.getKind()) {
                    case TYPE:
                        result.add(parameter.getTypeSignature());
                        break;
                    case NAMED_TYPE:
                        result.add(parameter.getNamedTypeSignature().getTypeSignature());
                        break;
                    default:
                        throw new IllegalStateException(
                                format("Expected all parameters to be of kind TYPE or NAMED_TYPE but [%s] kind was found for parameter: [%s]",
                                        parameter.getKind(), parameter));
                }
            }
            return result;
        }
    }

```